Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -111,9 +122,16 @@ public sealed class Server : ServerBase, IDisposable {
 
             return new InitializeResult {
                 capabilities = new ServerCapabilities {
-                    completionProvider = new CompletionOptions { resolveProvider = true },
                     textDocumentSync = new TextDocumentSyncOptions { openClose = true, change = TextDocumentSyncKind.Incremental },
-                    hoverProvider = true
+                    completionProvider = new CompletionOptions {
+                        triggerCharacters = new[] { "." },
+                        resolveProvider = true
+                    },
+                    hoverProvider = true,
+                    signatureHelpProvider = new SignatureHelpOptions { triggerCharacters = new[] { "(,)" } },
```

Comment:
Why does a close parenthesis trigger signature help?

Reply:
Usually nested calls as in `foo(bar()|` <- you may want signature help return to `foo`